### PR TITLE
Fixes frozen GUI when updating alert tree in non EDT

### LINKED ISSUE #5259: GUI hang (and NPE exceptions) when using Context Alert Filters
**Describe the bug**

ZAP UI & API freezes when running automated active scan. Suspect is `alertFilters` addon based on log.

My command to start ZAP UI:
```
java -XX:MetaspaceSize=128M -XX:MaxMetaspaceSize=256M  \
-jar /home/user1/ZAP_2.7.0/zap.7.0.jar \
-port 8081 \
-dir /home/user1/.ZAP \
-installdir /home/user1/ZAP_2.7.0 \
-config api.disablekey=true \
-config dirs=/var/www/myproject/nodirectaccess/cron/ProcessorServer/zap/scripts \
-config scanner.hostPerScan=2 \
-config scanner.threadPerHost=15 \
-config scanner.delayInMs=0 \
-config scanner.injectable=3 \
-config scanner.enabledRPC=63 \
-config scanner.scanHeadersAllRequests=false
```

First error in the logs (Full error is down below);

```
30264 [ZAP-PassiveScanner] ERROR org.zaproxy.zap.extension.alertFilters.ExtensionAlertFilters  - java.lang.NullPointerException
```

**To Reproduce**
Steps to reproduce the behavior (At least, this is how my automated flow works):
1.  Start ZAP UI (v2.7.0 with the command provided above)
2.  Run `core.deleteAllAlerts()`, `ascan.stopAllScans()` and `ascan.removeAllScans()`,
3.  Enable custom zest scripts,
4.  Run `core.newSession()`,
5.  Import ZAP contex.
6.  Run automated passive and active scans.

**Expected behavior**
ZAP UI doesn't freeze

**Screenshots**

**Software versions**
 - ZAP: 2.7.0
 - Add-on: [id=alertFilters, version=7.0.0]
 - OS: Ubuntu 18.04
 - Java: 
```
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
```
 - Browser: -

**Errors from the zap.log file**
https://pastebin.com/iYeMu0ps

Please shout how I can provide more details to reproduce this.

** Additional Info **

My context file - https://pastebin.com/2Dehw9vv
Thread dump output - https://pastebin.com/Wv0um8G8

### PR DESCRIPTION
**Pull request by chozzz**
This fixes #5259 by updating alert tree in a different thread when it's not in EDT

### CODE REVIEW

File: src/org/zaproxy/zap/extension/alert/ExtensionAlert.java [hunk 1]
```diff
@@ -436,6 +436,25 @@ public void updateAlertInTree(Alert originalAlert, Alert alert) {
         if (Constant.isLowMemoryOptionSet()) {
             return;
         }
+        
+        if (EventQueue.isDispatchThread()) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/alert/ExtensionAlert.java]: We should also do this if there's no view (e.g. `getView() == null`).

**Comment by psiinon** [comment]
@chozzz no need to close this...

**Comment by thc202** [comment]
I guess it was automatically closed when updating the `develop` branch from upstream.

**Comment by chozzz** [comment]
@psiinon Sorry, my local branching is screwed up.. I'll raise another PR for this..

**Pull request closed without merging**